core/src/main/java/org/mule/runtime/core/internal/context/DefaultMuleContext.java

```diff
@@ -414,6 +418,9 @@ public void stop() throws MuleException {
 
       final org.mule.runtime.api.artifact.Registry apiRegistry = getApiRegistry();
       listeners.forEach(l -> l.onStop(this, apiRegistry));
+      Optional<ArtifactStoppedListener> optionalArtifactStoppedListener =
+          Optional.ofNullable(getRegistry().lookupObject(ARTIFACT_STOPPED_LISTENER));
+      optionalArtifactStoppedListener.ifPresent(ArtifactStoppedListener::onStop);
```

Comment:
no need to use the intermediate `Optional` just for this. a simple null check will do.